src/core/utils/hooks/useCountdownTimer.ts

```diff
@@ -0,0 +1,41 @@
+import {useEffect, useRef, useState, useLayoutEffect} from "react";
+
+import {MINUTE_IN_S, SECOND_IN_MS} from "../time/timeConstants";
+import {CountDownResults} from "../time/timeTypes";
+import {calculateRemainingTime} from "../time/timeUtils";
+
+/**
+ * A React Hook that provides a countdown time
+ * @param {Date} target the target date we're counting down to
+ * @param {number} cadence the rate of the timer in seconds
+ *
+ * @returns {object} the CountDownResults object with ref
+ */
+function useCountDownTimer(target: Date, cadence = MINUTE_IN_S): CountDownResults {
+  const interval = useRef<NodeJS.Timeout>();
+  const [countdown, setCountDown] = useState<CountDownResults>(
+    calculateRemainingTime(target)
+  );
+
+  useLayoutEffect(() => {
+    interval.current = setInterval(() => {
+      if (countdown.delta >= 0) {
```

Comment:
Why are we comparing with the previous value of countdown ? We should calculate the new "remainingTime" data and compare with that
```suggestion
      const data = calculateRemainingTime(target);
      
      if (data.delta >= 0) {
          setCountDown(data);
```